dali/core/tensor_shape_test.cc

```diff
@@ -1316,5 +1316,50 @@ TEST(TensorListShapeTest, SampleRangeTest) {
   EXPECT_EQ(sample_range(TensorListShape<>(), 0, 0), TensorListShape<>());
 }
 
+
+TEST(TestListShapeTest, UnfoldOuterDimTest_static) {
+  TensorListShape<4> tls_rank4 = {{{1, 2, 3, 4}, {5, 6, 7, 8}}};
+  TensorListShape<1> tls_rank1 = {{{1}, {5}}};
+
+  TensorListShape<> ref_for_r4 = {
+      {{2, 3, 4}, {6, 7, 8}, {6, 7, 8}, {6, 7, 8}, {6, 7, 8}, {6, 7, 8}}};
+  TensorListShape<0> ref_for_r1 = {{{}, {}, {}, {}, {}, {}}};
+
+  EXPECT_EQ(unfold_outer_dim(tls_rank4), ref_for_r4);
+  EXPECT_EQ(unfold_outer_dim(tls_rank1), ref_for_r1);
+  EXPECT_EQ(unfold_outer_dim<3>(tls_rank4), ref_for_r4);
+  EXPECT_EQ(unfold_outer_dim<0>(tls_rank1), ref_for_r1);
+
+  constexpr bool ret_type_00 =
+      std::is_same<decltype(unfold_outer_dim<0>(tls_rank1)), TensorListShape<0>>::value;
+  EXPECT_EQ(ret_type_00, true);
```

Comment:
```suggestion
  EXPECT_TRUE(ret_type_00);
```
is generally preferred